### PR TITLE
feat: add policy metrics

### PR DESCRIPTION
**Pull request by naidenovaleksei**
Validation-time metrics for the policy objective, split by action-based cluster

**1. New prediction keys**
Six new keys added to ObjectivePredictionKey and implemented in PolicyObjective.predict():
- `score_l1_rel`: L1 error relative to the ground-truth magnitude (|pred - gt| / (gt + ε))
- `score_signed_error`: signed error (pred - gt), useful for detecting systematic bias
- `prediction_diff_prev` / `ground_truth_diff_prev`: delta from the previous timestep for both prediction and ground truth
- `prediction_diff_hist` / `ground_truth_diff_hist`: delta from the episode history mean/mode (continuous/discrete respectively)

**2. PredictMetricsCallback**
New callback that:
- Temporarily swaps the model's `PredictionConfig` at the start of each validation epoch to compute a configurable set of objectives
- Accumulates per-batch results and logs epoch-level means to W&B under the predict/ prefix
- Optionally accepts a `cluster_fn` and a cluster_metrics map to log per-cluster metrics (e.g. predict/braking/score_l1/...)

**3. RuleBasedCluster**
New utility that assigns a named action-based scenario label to each episode in a batch using priority-ordered rules over reduced batch fields. Supported reduce ops: last, last_diff. Supported condition ops: ge, lt, abs_ge, abs_lt.

**Config changes:**
- config/cluster_fn/rule_based.yaml - defines 8 driving scenarios:
    - highway
    - braking_turn
    - braking
    - cruise_turn
    - acceleration
    - gas_release
    - cruise
    - idle_coast (default)
- config/trainer/callbacks/finetune.yaml — wires in PredictMetricsCallback with per-cluster metric tracking
- config/experiment/yaak/control_transformer/finetune.yaml — adds cluster_fn: rule_based to the finetune experiment
- config/inference/yaak/control_transformer/policy.yaml — exposes new prediction keys for inference output
Rerun logger template updated to include the new prediction keys

**Tests**
Three new tests covering: metric prefix, per-cluster breakdown, and prediction config swap/restore behavior.

### CODE REVIEW

File: config/experiment/yaak/control_transformer/finetune.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ defaults:
   - /datamodule: yaak/train
   - /trainer: default
   - /trainer/callbacks: finetune
+  - /cluster_fn: rule_based
```
Comment (koritsky) [inline config/experiment/yaak/control_transformer/finetune.yaml]: not sure if it should be defined at that level. it is read at `config/trainer/callbacks/*.yaml`, can config path be defined directly there?

File: src/rmind/callbacks/predict_metrics.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    def on_validation_batch_end(
+        self,
+        trainer: pl.Trainer,
+        pl_module: pl.LightningModule,
+        outputs: Any,
+        batch: Any,
+        batch_idx: int,
+        dataloader_idx: int = 0,
+    ) -> None:
+        predictions = pl_module.predict_step(batch)
```
Comment (koritsky) [inline src/rmind/callbacks/predict_metrics.py:56]: yeah, at this point calculating metrics in `predict_step` is a bit misleading. can metrics logic be abstracted into a separate module?

File: src/rmind/components/objectives/policy.py [hunk 3]
```diff
@@ -299,4 +305,171 @@ def fn(
...
+                    action_type: tuple[Modality, str], x: torch.Tensor
+                ) -> torch.Tensor:
+                    gt_episode = cast(
+                        "Tensor", episode.input[action_type]
+                    )  # (b, ep_length, 1)
+                    gt = gt_episode[:, -1]  # (b, 1)
+                    match action_type:
+                        case (Modality.CONTINUOUS, _):
+                            prediction = x[..., 0]  # (b, 1)
+                            return (prediction - gt).abs() / (gt + 1e-4)
```
Comment (koritsky) [inline src/rmind/components/objectives/policy.py]: when `gt < 0` (negative steering angle), L1 score becomes negative. is it intentional? if not, `gt.abs()` might be better

File: src/rmind/callbacks/predict_metrics.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            else None
+        )
+
+        for key, v in predictions.flatten_keys("/").items():
+            if not isinstance(v, Tensor):
+                continue
+            if all_allowed is not None and key not in all_allowed:
+                continue
+
+            self._accumulated[key].append(v.float().mean())
```
Comment (koritsky) [inline src/rmind/callbacks/predict_metrics.py]: we accumulate per batch mean and hence metrics vary based on batch size

File: config/inference/yaak/control_transformer/policy.yaml [hunk 5]
```diff
@@ -43,6 +48,12 @@ trainer:
         - [policy, prediction_probs, value]
         - [policy, score_l1, value]
         - [policy, score_logprob, value]
+        - [policy, score_l1_rel, value]
+        - [policy, score_signed_error, value]
```
Comment (koritsky) [inline config/inference/yaak/control_transformer/policy.yaml:54]: `score_signed_error` is selected here, but is missing at `config/trainer/callbacks/finetune.yaml`

**Review by koritsky — CHANGES_REQUESTED** [review]
minor issues, otherwise good for me

it would be nice to rethink the whole prediction codebase, but up to you. not sure if its urgent for now, yet it's a nice time since context is fresh in your head

File: config/experiment/yaak/control_transformer/finetune.yaml (hunk 1 above)
Comment (naidenovaleksei) [reply config/experiment/yaak/control_transformer/finetune.yaml]: fixed in 408a5d6b0a49b9d320b2fd2f0e0900e822abd085

File: src/rmind/components/objectives/policy.py (hunk 3 above)
Comment (naidenovaleksei) [reply src/rmind/components/objectives/policy.py]: fixed in 0e2f4ffcc75148c6bf4e3a513d557534d031f950

File: src/rmind/callbacks/predict_metrics.py (hunk 4 above)
Comment (naidenovaleksei) [reply src/rmind/callbacks/predict_metrics.py]: fixed in 0e2f4ffcc75148c6bf4e3a513d557534d031f950

File: config/inference/yaak/control_transformer/policy.yaml (hunk 5 above)
Comment (naidenovaleksei) [reply config/inference/yaak/control_transformer/policy.yaml:54]: fixed in 0e2f4ffcc75148c6bf4e3a513d557534d031f950

File: src/rmind/callbacks/predict_metrics.py (hunk 2 above)
Comment (naidenovaleksei) [reply src/rmind/callbacks/predict_metrics.py:56]: Since the objective and metrics are tightly coupled, we decided to come back to the refactoring question when we face a new objective or a new concept in action generation

**Review by koritsky — APPROVED** [review]
